src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -130,6 +132,15 @@ public HttpConnectionPoolManager(HttpConnectionSettings settings)
                     _proxyCredentials = _proxy.Credentials ?? settings._defaultProxyCredentials;
                 }
             }
+
+            _heartBeatTimer = new Timer(static state =>
```

Comment:
If there is no ping configured then there is no need for this timer. Only create when pings are enabled?